auth/src/main/java/com/firebase/ui/auth/util/SmartlockUtil.java

```diff
@@ -12,44 +12,54 @@
  * Helper class to deal with Smartlock Flows.
  */
 public class SmartlockUtil {
-
     /**
      * If SmartLock is enabled and Google Play Services is available, start the save credential
      * Activity. Otherwise, finish the calling Activity with RESULT_OK.
-     * @param activity the calling Activity.
-     * @param requestCode request code to use when starting the save operation.
-     * @param parameters calling Activity flow parameters.
+     *
+     * @param activity     the calling Activity.
+     * @param requestCode  request code to use when starting the save operation.
+     * @param data         An Intent to be merged. Must be a {@code new Intent().putExtras(...)}.
```

Comment:
Ok I just looked through some Android source and here's whats happening.  When you merge the extras of two bundles (say `dst.Extras(src)`) it basically does this (pseudocode):

```
for (String key : src.getExtras().getKeys()) {
  dst.putExtra(key, src.getExtra(key));
}
```

This means for a given `key` if `src.getExtra(key)` is `null` then it will erase the extra with the same key in `dst`.  I think that's why your email value is disappearing.

Maybe you want to create a utility method to merge extras only if they're non-null?  I don't want to reverse the order of operations here just to get around the bug, as you said it makes the result of the intent counter-intuitive.